app/src/main/java/com/github/se/polyfit/ui/components/selector/MealSelector.kt

```diff
@@ -0,0 +1,162 @@
+import com.github.se.polyfit.ui.theme.SecondaryGrey
+
+@Composable
+fun MealSelector(
+    selectedMeal: Meal,
+    setSelectedMeal: (Meal) -> Unit,
+    meals: List<Meal>,
+    setPostMeal: (Meal) -> Unit
+) {
+  val context = LocalContext.current
+  var showMealSearch by remember { mutableStateOf(false) }
+  val isDisplayMealDetails by
+      remember(selectedMeal) { derivedStateOf { selectedMeal.isComplete() } }
+
+  Box(modifier = Modifier.testTag("MealSelector")) {
+    if (!showMealSearch) {
+      Row(
+          modifier =
+              Modifier.padding(start = 12.dp, top = 20.dp)
+                  .fillMaxWidth()
+                  .testTag("MealSelectorRow")
+                  .clickable {
+                    showMealSearch = true
+                    setSelectedMeal(Meal.default())
+                  }) {
+            Icon(
+                imageVector = Icons.Filled.Star,
+                contentDescription = "Star",
+                tint = SecondaryGrey,
+                modifier = Modifier.padding(end = 20.dp))
+
+            if (!isDisplayMealDetails) {
+              Text(
+                  text = context.getString(R.string.shareRecipePlaceholder),
+                  color = SecondaryGrey,
+                  fontSize = 20.sp,
+              )
+            } else {
+              Text(
+                  text = selectedMeal.name,
+                  fontSize = 20.sp,
+                  modifier = Modifier.weight(1f),
+              )
+              Text(
+                  text = "${selectedMeal.calculateTotalCalories()} kcal",
+                  fontSize = 20.sp,
+                  modifier = Modifier.weight(1f).padding(end = 20.dp),
+                  textAlign = TextAlign.End,
+              )
+            }
+          }
+    } else {
+      MealInputTextField(
+          meals = meals,
+          onCollapseSearchBar = { showMealSearch = false },
+          onMealChange = {
+            setSelectedMeal(it)
+            setPostMeal(it)
+          })
+    }
+    if (isDisplayMealDetails) {
+      MealDetails(selectedMeal, Modifier.padding(top = 10.dp))
+    }
+  }
+}
+
+@Composable
+fun MealDetails(meal: Meal, modifier: Modifier = Modifier) {
+  Box(modifier = modifier.fillMaxWidth().testTag("MealDetails")) {
+    Column(modifier = Modifier.fillMaxWidth(0.75f).align(Alignment.Center)) {
+      LazyColumn(modifier = Modifier.heightIn(max = 200.dp)) {
+        items(meal.ingredients) { ingredient ->
+          Row(
+              modifier = Modifier.fillMaxWidth().padding(4.dp).testTag("Ingredient"),
+              verticalAlignment = Alignment.CenterVertically) {
+                Text(
+                    text = ingredient.name,
+                    modifier = Modifier.weight(1f).padding(start = 10.dp),
+                    fontSize = 16.sp,
+                    color = SecondaryGrey)
+
+                Text(
+                    text = "${ingredient.amount} ${ingredient.unit.toString().lowercase()}",
```

Comment:
You can use resource strings for this too, and pass in params

Reply:
I think its good for static resource but this is just two variable together, so it will be ok